Add #normalizeCamelCase and test

hernanmd opened the pull request:
Add method to split a camel cased String into its "normalized" string version, separated by spaces.

```smalltalk
'PinkFloyd' normalizeCamelCase >>> 'Pink Floyd'
```

Commit:
Add #normalizeCamelCase and test

astares commented on src/Collections-Strings-Tests/StringTest.class.st:
```diff
@@ -1654,6 +1654,17 @@ StringTest >> testMatch [
+	self assert: 'CamelCase123' normalizeCamelCase equals:'Camel Case123'.
+	self assert: 'ABC' normalizeCamelCase equals: 'ABC'.
+	self assert: 'TresTristesTigres' normalizeCamelCase equals: 'Tres Tristes Tigres'.
+	self assert: 'TresTristesTigresABC' normalizeCamelCase equals: 'Tres Tristes Tigres ABC'
```
please add at least one test with two wording parts as input **separated by space** to cover that case

self assert: 'CamelCase HelloWorld' normalizeCamelCase equals: ...

hernanmd replied:
Added

astares commented on src/Collections-Strings/String.class.st:
```diff
@@ -2178,6 +2178,19 @@ String >> match: text [
+	'TheRollingStones' normalizeCamelCase >>> 'The Rolling Stones'
+	"
+	^ self class streamContents: [ :out |
+		self do: [ :e |
```
why do you call the block parameter "out" and not "s" (like Stream) or better "stream" 
why do you call the block parameter "e" and not "c" (like Character) or better "char" or even better "character"

hernanmd replied:
Ok, good suggestions, I just integrated them and fixed the case to not write two consecutive blank spaces where there is a single space in the input string.

Commit:
Merge 7cc434c26aac6bf6b4cec980b7ad0d84af4ea268

Commit:
Handle the case where there is a space to not write two spaces in the output stream.
Add test case as suggested by @astares : self assert: 'CamelCase HelloWorld' normalizeCamelCase equals:...
Call the block parameter "char" instead of "e" as suggested by @astares
